cost/scheduled_reports/scheduled_report.pt

```diff
@@ -0,0 +1,306 @@
+      // start_at from beginning of this month
+      start_at = getFormattedDailyDate(new Date(now.getFullYear(), now.getMonth(), 1))
+      // end_at to end of this month
+      end_at = getFormattedDailyDate(now)
+    } else if (range === "monthly" && scale==="previous") {
+      // start_at from beginning of last month
+      start_at = getFormattedDailyDate(new Date(now.getFullYear(), now.getMonth() - 1, 1))
+      // end_at to end of last month
+      end_at = getFormattedDailyDate(new Date(now.setDate(0)))
+    }
+
+
+    var body = {
+      "dimensions":[]
+      "granularity":"day",
+      "start_at": start_at ,
+      "end_at": end_at
+      "metrics":[cost_metric[param_cost_metric]],
+      "billing_center_ids":param_billing_centers
+      }
+
+    var request = {
+      auth:  'auth_rs',
+      //host:  rs_optima_host,
+      host: "front-service-b.indigo.rightscale.com",
+      scheme: 'https',
+      verb: 'POST',
+      path:  "/bill-analysis/orgs/"+ rs_org_id + "/costs/aggregated",
+      headers: {
+        "API-Version": "1.0",
+        "Content-Type":"application/json"
+      },
+      body: JSON.stringify(body)
+    }
+  EOS
+end
+
+datasource "ds_report" do
+  run_script $js_report,$current_week_costs,$previous_week_costs,$current_month_costs,$previous_month_costs,$param_cost_metric
+end
+
+script "js_report", type: "javascript" do
+  parameters "current_week_costs","previous_week_costs","current_month_costs","previous_month_costs","param_cost_metric"
+  result "report"
+  code <<-EOS
+      // create a map to use in the request metric field.
+      var cost_metric = {
+        "Unamortized Unblended":"cost_nonamortized_unblended_adj",
+        "Amortized Unblended":"cost_amortized_unblended_adj",
+        "Unamortized Blended": "cost_nonamortized_blended_adj",
+        "Amortized Blended":"cost_amortized_blended_adj"
+      }
+
+      var metric = cost_metric[param_cost_metric]
+      var current_week_total = 0.00
+      for (var i = 0; i < current_week_costs['rows'].length; i++) {
+        var row =  current_week_costs['rows'][i]
+        current_week_total += row['metrics'][metric]
+      }
+
+      var previous_week_total = 0.00
+      for (var i = 0; i < previous_week_costs['rows'].length; i++) {
+        var row =  previous_week_costs['rows'][i]
+        previous_week_total += row['metrics'][metric]
+      }
+
+      var previous_month_total = 0.00
+      for (var i = 0; i < previous_month_costs['rows'].length; i++) {
+        var row =  previous_month_costs['rows'][i]
+        previous_month_total += row['metrics'][metric]
+      }
+
+      var current_month_total = 0.00
+      for (var i = 0; i < current_month_costs['rows'].length; i++) {
+        var row =  current_month_costs['rows'][i]
+        current_month_total += row['metrics'][metric]
+      }
+
+      var weekly_avg = 0
+      if (current_week_costs['rows'].length > 0){
+        //weekly_avg = current_week_total / current_week_costs['rows'].length
```

Comment:
Should probably remove commented out code.

Reply:
removed